### PR TITLE
feat(cli): suggest matching commands

### LINKED ISSUE #476: Suggest matching commands if the user makes a typo
## Summary

Suggest matching commands if the user mistypes.

### Basic example

`gridsome serv` results in the following prompt:-
```md
Did you mean serve?
```

### Motivation

Why are we doing this? What use cases does it support? What is the expected outcome?
As per what I think it is a worth feature to be included in the next release.

### PR DESCRIPTION
**Pull request by jamesgeorge007**
Fixes #476 

Suggests matching commands if the user mistypes.

### CODE REVIEW

**Comment by jamesgeorge007** [comment]
@robaxelsen Thoughts :thinking:

File: packages/cli/bin/gridsome.js [hunk 1]
```diff
@@ -3,13 +3,26 @@
...
 
+function suggestCommands (cmd) {
+  const availableCommands = program.commands.map(cmd => {
+    return cmd._name
+  })
+
+  const suggestion = didYouMean(cmd, availableCommands)
+  if (suggestion) {
+    console.log()
+    console.log(`Did you mean ${suggestion}`)
```
Comment (hjvedvik) [inline packages/cli/bin/gridsome.js]:
```suggestion
    console.log(`Did you mean ${suggestion}?`)
```

**Comment by jamesgeorge007** [comment]
@hjvedvik There are couple of issues with my local environment specifically relating to `yarn`.

**Comment by jamesgeorge007** [comment]
@hjvedvik Hope we're good to go :+1:

**Comment by hjvedvik** [comment]
Looks good now. Thanks again @jamesgeorge007 :)

**Pull request merged**